firmware/lucidgloves-firmware/src/Main.cpp

```diff
@@ -164,18 +120,52 @@ void loop() {
       
     }
 
-    comm->output(encode(fingerPosCopy, getJoyX(), getJoyY(), joyButton, triggerButton, aButton, bButton, grabButton, pinchButton, calibButton, menuButton));
+    comm->output(encoding->encode(fingerPosCopy, input.getJoyX(), input.getJoyY(), joyButton, triggerButton, aButton, bButton, grabButton, pinchButton, calibButton, menuButton));
     #if USING_FORCE_FEEDBACK
       char received[100];
       if (comm->readData(received)){
         int hapticLimits[5];
         //This check is a temporary hack to fix an issue with haptics on v0.5 of the driver, will make it more snobby code later
         if(String(received).length() >= 5) {
-           decodeData(received, hapticLimits);
-           writeServoHaptics(hapticLimits); 
+           DecodedData recievedData = encoding->decodeData(received);
+           haptics.writeServoHaptics(recievedData.servoValues); 
+
+           if (recievedData.fields.specialCommandReceived){
+              if (recievedData.command == "ClearData")
+                input.clearFlags();
+              else if (recievedData.command == "SaveInter")
+                input.saveIntermediate();
+              else if (recievedData.command == "SaveTravel")
+                input.saveTravel();
+           }
```

Comment:
`ClearData`, `SaveInter`, `SaveTravel` are also strings defined elsewhere, having variables for the strings might help to avoid typos in the future.